overlord/overlord_test.go

```diff
@@ -1315,3 +1317,42 @@ func (ovs *overlordSuite) TestStartupTimeout(c *C) {
 	c.Check(to, Equals, (30+5+5)*time.Second)
 	c.Check(reasoning, Equals, "pessimistic estimate of 30s plus 5s per snap")
 }
+
+func (ovs *overlordSuite) TestLockWithTimeoutHappy(c *C) {
+	f, err := ioutil.TempFile("", "testlock-*")
+	defer func() {
+		f.Close()
+		os.Remove(f.Name())
+	} ()
+	c.Assert(err, IsNil)
+	flock, err := osutil.NewFileLock(f.Name())
+	c.Assert(err, IsNil)
+
+	err = overlord.LockWithTimeout(flock, time.Second)
+	c.Check(err, IsNil)
+}
+
+func (ovs *overlordSuite) TestLockWithTimeoutFailed(c *C) {
+	f, err := ioutil.TempFile("", "testlock-*")
+	defer func() {
+		f.Close()
+		os.Remove(f.Name())
+	} ()
+	c.Assert(err, IsNil)
+	flock, err := osutil.NewFileLock(f.Name())
+	c.Assert(err, IsNil)
+
+	cmd := exec.Command("flock", "-w", "2", f.Name(), "-c", "echo acquired && sleep 1")
+	stdout, err := cmd.StdoutPipe()
+	c.Assert(err, IsNil)
+	err = cmd.Start()
```

Comment:
we're missing a cmd.Wait() somewhere

Reply:
Yes and no :-) I don't want to add it, otherwise this test will always take at least a second to complete. I was looking for a way to kill the process instead, but I didn't find it. Ah! I now see that there's a `os.Process` member in `cmd`. I'll kill it :-)

Reply:
Done!